script.js

```diff
@@ -1,12 +1,363 @@
+      readMore.style.display = "none";
+    }
+    
+    //An event listener to expand the summary text
+    readMore.addEventListener('click', () => {
+      expandingList.removeChild(expandingList.lastChild);
+      expandingList.appendChild(span);
+      span.setAttribute('class', 'more-text summary');
+      span.style.display = 'flex';
+      span.appendChild(readLess);
+
+    })
+    
+    //An event listener to collapse the summary text
+    readLess.addEventListener('click', () => {
+      expandingList.removeChild(expandingList.lastChild);
+      let paragraph = document.createElement('p');
+      expandingList.append(paragraph);
+      paragraph.setAttribute('class', 'summary')
+      paragraph.innerHTML = `${truncatedText}...`;
+      paragraph.appendChild(readMore);
+      span.style.display = 'none';
+
+    })
+    
+    //Select a show options
+    let selectShows = document.querySelector('#shows');
+    let options = document.createElement('option');
+    options.innerHTML = `${element.name}`;
+    selectShows.appendChild(options);
+    let anchorTag = document.createElement('a');
+    anchorTag.setAttribute('href', '#');
+    options.appendChild(anchorTag);
+
+    // An event listener to fetch a show's episodes data when selected
+    selectShows.addEventListener('change', function(){
+
+      displayShows.style.display = 'none';
+      displayEpisodes.style.display = 'block';
+      
+      if (`${element.name}` === this.value) { 
+          `${element.id}`;
+          showID =`${element.id}`;
+          loadEpisodes();
+          navLink.style.display = 'block';
+          document.getElementById('show-episodes').innerHTML = `Select from the list of ${element.name} episodes`;
+          document.getElementById('show-name').innerHTML = `${element.name}`;
+          searchForEpisodes.style.display = 'block';
+      }
+
+    })
+
+  });
+ 
+}
+
+
+// An event listener wrapped inside a function to dynamically update the web page while a user is typing in the search bar.
+function showSearch () { 
+
+  const searchInfo = document.querySelector('.search-info');
+  const searchInfo2 = document.querySelector('.search-info2');
+  
+  searchBar.addEventListener('keyup', (e) => {
+
+    // Revealing the hidden information lines
+    searchInfo.style.display = "block";
+    searchInfo2.style.display = "block";
+    
+    //Filtering search results
+    let searchValue = e.target.value.toLowerCase();
+   
+    let searchResult = data.filter((element) => {
+      return (element.name.toLowerCase().includes(searchValue) || element.summary.toLowerCase().includes(searchValue) || element.genres.toString().toLowerCase().includes(searchValue));
+    })  
+    while(rootShows.firstChild) {
+      rootShows.removeChild(rootShows.firstChild)
+    }
+    allShows(searchResult)
+  
+    if (searchValue === "") {
+      searchInfo.style.display = "none";
+      searchInfo2.style.display = "none";
+      window.location.reload();
```

Comment:
I would avoid using a window reload to reset the page, it would probably be best to reset things using a function in your code